src/Build.OM.UnitTests/Construction/ProjectTargetElement_Tests.cs

```diff
@@ -339,6 +343,42 @@ public void SetReturns()
             Assert.True(project.HasUnsavedChanges);
         }
 
+        /// <summary>
+        /// Parse invalid property under target
+        /// </summary>
+        [Fact]
+        public void ReadInvalidPropertyUnderTarget()
+        {
+            using (TestEnvironment env = TestEnvironment.Create())
+            {
+                ChangeWaves.ResetStateForTests();
+                env.SetEnvironmentVariable("MSBUILDDISABLEFEATURESFROMVERSION", ChangeWaves.Wave17_6.ToString());
+                BuildEnvironmentHelper.ResetInstance_ForUnitTestsOnly();
+
+                var error= Assert.Throws<InvalidProjectFileException>(() =>
+                {
+                    string projectFile = @"
+                    <Project>
+                        <Target Name='t'>
+                            <test>m</test>
+                        </Target>
+                    </Project>";
+
+                    TransientTestFile file = env.CreateFile("proj.csproj", projectFile);
+                    ProjectCollection collection = new ProjectCollection();
+                    collection.LoadProject(file.Path).Build().ShouldBeTrue();
+                });
```

Comment:
Because of a long-ago automated conversion, many of the methods in this file follow a bad practice for `Assert.Throws`. Ideally it should be wrapped around just the statement we expect to throw:

```suggestion
                string projectFile = @"
                <Project>
                    <Target Name='t'>
                        <test>m</test>
                    </Target>
                </Project>";

                TransientTestFile file = env.CreateFile("proj.csproj", projectFile);
                ProjectCollection collection = new ProjectCollection();

                var error = Assert.Throws<InvalidProjectFileException>(() =>
                    collection.LoadProject(file.Path).Build().ShouldBeTrue());
```

Reply:
Thanks. Done